sdks/python/apache_beam/io/filebasedsource.py

```diff
@@ -377,8 +382,13 @@ def process(self, element, *args, **kwargs):
     if not source_list:
       return
     source = source_list[0].source
+
+    data = []
     for record in source.read(range.new_tracker()):
-      yield record
+      data.append(record)
```

Comment:
I think this could result in OOMs. There's not size limit to the number of records that can be read here so putting all records in an array is not a viable option.

May be just do a yield of file name and data ?